crates/aptos/src/update/movefmt.rs

```diff
@@ -0,0 +1,195 @@
+                let stdout = String::from_utf8(output.stdout)
+                    .context("Failed to parse current version of movefmt as UTF-8")?;
+                let version = extract_movefmt_version(&stdout);
+                if !version.is_empty() {
+                    let current_version = version.trim_start_matches("movefmt v").to_string();
+                    Some(current_version)
+                } else {
+                    None
+                }
+            },
+            Err(_) => None,
+        };
+
+        // Strip v prefix from target version if present.
+        let target_version = self.target_version.trim_start_matches('v').to_string();
+
+        Ok(UpdateRequiredInfo {
+            current_version,
+            target_version,
+        })
+    }
+
+    fn build_updater(&self, info: &UpdateRequiredInfo) -> Result<Box<dyn ReleaseUpdate>> {
+        // Determine the target we should download based on how the CLI itself was built.
+        let arch_str = get_arch();
+        let build_info = cli_build_information();
+        let target = match build_info.get(BUILD_OS).context("Failed to determine build info of current CLI")?.as_str() {
+            "linux-aarch64" | "linux-x86_64" => "unknown-linux-gnu",
+            "macos-aarch64" | "macos-x86_64" => "apple-darwin",
+            "windows-x86_64" => "windows",
+            wildcard => bail!("Self-updating is not supported on your OS ({}) right now, please download the binary manually", wildcard),
+        };
+
+        let target = format!("{}-{}", arch_str, target);
+
+        let install_dir = match self.install_dir.clone() {
+            Some(dir) => dir,
+            None => {
+                let dir = get_additional_binaries_dir();
+                // Make the directory if it doesn't already exist.
+                std::fs::create_dir_all(&dir)
+                    .with_context(|| format!("Failed to create directory: {:?}", dir))?;
+                dir
+            },
+        };
+
+        let current_version = match &info.current_version {
+            Some(version) => version,
+            None => "0.0.0",
+        };
+
+        Update::configure()
+            .bin_install_dir(install_dir)
+            .bin_name(FORMATTER_BINARY_NAME)
+            .repo_owner(&self.repo_owner)
+            .repo_name(&self.repo_name)
+            .current_version(current_version)
+            .target_version_tag(&format!("v{}", info.target_version))
+            .target(&target)
+            .build()
+            .map_err(|e| anyhow!("Failed to build self-update configuration: {:#}", e))
+    }
+}
+
+#[async_trait]
+impl CliCommand<String> for FormatterUpdateTool {
+    fn command_name(&self) -> &'static str {
+        "UpdateMovefmt"
+    }
+
+    async fn execute(self) -> CliTypedResult<String> {
+        update_binary(self).await
+    }
+}
+
+#[cfg(target_arch = "x86_64")]
+fn get_arch() -> &'static str {
+    "x86_64"
+}
+
+#[cfg(target_arch = "aarch64")]
```

Comment:
I think sometimes it can be `arm` too, are we certain it is always only aarch64.

Reply:
yes, it can be `arm`. What is different between `aarch64` and `arm` though?

Reply:
Not sure, but we should handle both if we expect some systems to have a target arch of `arm`.

Reply:
it seems that `arm` is 32bit and mac os only supports 64bit. Does Aptos CLI support linux with 32bit arm architecture?